### ISSUE TITLE
send pubkeys as attachments instead of inline

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-browser#490]
This should help enigmail recognize it

**Comment by hommehabile** [comment]
Also, GPG for Outlook is not currently able to handle inline public keys. It breaks the message, as seen on this image:
![image](https://user-images.githubusercontent.com/26507295/34215041-2ccf9d52-e572-11e7-8ffa-8155b5717184.png)
And according to this post [https://dev.gnupg.org/T3531](url), it is still preferable to attach the public key as a file.

**Comment by tomholub** [comment]
Good to know. In fact, Protonmail has the same issue

**Comment by fubarslayer** [comment]
Aside from the issue it's causing I don't understand why attaching the public key is default behaviour.
This isn't the case in for example mailvelope and most of the other software allows you to at least alter the default behaviour.
I have a contact who uses flowcrypt but doesn't want his public key to end up on a key server (flowcrypt has an option not to send his public key to attester.flowcrypt.com so that's ok).
He also doesn't want his public key to be attached to his emails but there is no option to change the default behaviour so for each and every email he sends he has to keep in mind to click on "PK".

**Comment by tomholub** [comment]
If he doesn't have public key on key servers, and doesn't want to send it by email, how does he deliver the public keys to his contacts?

**Comment by fubarslayer** [comment]
Let me clarify, this is not about not wanting to send his public key by email.
This is about the fact that there's no need to automatically keep attaching his public key to every email after public keys have already been exchanged.
When I started using pgp years ago I also attached my public key to every email but I stopped doing that because some of my contacts got confused by it.
They sent me messages to ask what the asc file was for, what they had to do with it and if they had to replace my public key in their email software with the public key attached to my email,...
When my contacts forwarded my emails to their contacts some of those people got even more confused if my public key was still attached.
If you, like my contact, don't want your public key to end up with people you don't know or end up on key servers then attaching your public key to every email is certainly not the way to go.
I understand that you chose this as the default behaviour to make it easy for people to get started with pgp but I also believe there are valid reasons to provide an option to change the default behaviour.

**Comment by tomholub** [comment]
Got it. Actually, FlowCrypt should be checking if public key was already sent to that person before. If you let it send the pubkey one time, it should not prompt you to send it to that person second time.

Also, it should not generally be automatically attaching pubkeys to people who also use FlowCrypt.

Also, if your own pubkey is wrongly attested or wrongly recorded, it should start attaching it with every email. See FlowCrypt Settings -> Additional Settings -> Attester and double check it's all showing in green.

In other words, there is a little bit of magic involved. We're trying our best to find the right balance - most errors usually happen because a public key is wrong or missing somewhere, rather than sent too many times.

**Comment by tomholub** [comment]
ref https://mail.google.com/mail/u/0/#inbox/161bb15d29f0e842

**Comment by tomholub** [comment]
ref https://mail.google.com/mail/u/0/#inbox/161e178339be0257

**Comment by tomholub** [comment]
> Hi,
> 
> unfortunately Flowcrypt neither is conforming to the standard of
> PGP/INLINE RFC2440 6.2 nor to modern PGP/MIME RFC3156. They're
> incompatible to any other solution by ignoring the standards.
> 
> Yours,
> VB.

ref https://mail.google.com/mail/u/0/#inbox/1629bf450d118024

**Comment by tomholub** [comment]
This is now resolve and will ship in version 5.6.5 tonight.
